src/MSALWrapper/ExceptionListToStringConverter.cs

```diff
@@ -6,46 +6,48 @@ namespace Microsoft.Authentication.MSALWrapper
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using Microsoft.Identity.Client;
 
     /// <summary>
     /// The exception list to string converter.
     /// </summary>
     public static class ExceptionListToStringConverter
     {
         /// <summary>
-        /// Executes the convertion of exceptions to a string.
+        /// Extracts correlation IDs from the list of exceptions.
         /// </summary>
-        /// <param name="exceptions">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
-        public static string Execute(IEnumerable<Exception> exceptions)
+        /// <param name="exceptions">List of exceptions from which correlation IDs are extracted.</param>
+        /// <returns>List of correlation IDs.</returns>
+        public static List<string> ExtractCorrelationIDsFromException(IList<Exception> exceptions)
         {
-            if (exceptions == null || exceptions.Count() == 0)
+            var correlationIDs = new List<string>();
+
+            if (exceptions == null)
             {
                 return null;
             }
 
-            return string.Join("\n", exceptions.Select(SingleLineException));
-        }
-
-        /// <summary>
-        /// Converts exceptions to a single string.
-        /// </summary>
-        /// <param name="ex">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
-        private static string SingleLineException(Exception ex)
-        {
-            if (ex != null)
-            {
-                var message = ex.Message
-                                .Replace("\n", string.Empty)
-                                .Replace("\r", string.Empty)
-                                .Replace("\t", string.Empty);
-                return $"{ex.GetType()}: {message}";
-            }
-            else
+            foreach (Exception exception in exceptions)
             {
-                return "null";
+                if (exception.GetType() == typeof(MsalServiceException))

```

Comment:
`MsalServiceException` and `MsalUiRequiredException` are the only exceptions we collect correlation IDs for? Is that on purpose since we do catch exceptions such as `MsalClientException`, etc.

Reply:
Yes, those are the only exceptions with correlation IDs.